lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -91,37 +103,202 @@ Function CheckModified-File($module, $url, $dest, $headers, $credentials, $timeo
+
+    $webRequest = GetWebRequest -url $url -headers $headers -credentials $credentials -timeout $timeout -use_proxy $use_proxy -proxy $proxy
+
     if ($webRequest -is [System.Net.FtpWebRequest]) {
-        $webRequest.Method = [System.Net.WebRequestMethods+Ftp]::GetDateTimestamp
+        $webRequest.Method = [System.Net.WebRequestMethods+Ftp]::DownloadFile
     } else {
-        $webRequest.Method = [System.Net.WebRequestMethods+Http]::Head
+        $webRequest.Method = [System.Net.WebRequestMethods+Http]::Get
     }
 
     # FIXME: Split both try-statements and single-out catched exceptions with more specific error messages
     Try {
-        $webResponse = $webRequest.GetResponse()
-        $webLastMod = $webResponse.LastModified
+         $webResponse = $webRequest.GetResponse()
+         $responseStream = $webResponse.GetResponseStream()
+         $readStream = New-Object System.IO.StreamReader $responseStream
+         $web_checksum=$readStream.ReadToEnd()
+
+         $basename = (Split-Path -Path $([System.Uri]$src_file_url).LocalPath -Leaf)
+         $basename = [regex]::Escape($basename)
+         $web_checksum_str = $web_checksum -split '\r?\n' | Select-String -Pattern $("\s+\.?\/?\\?" + $basename + "\s*$")
+         if (-not $web_checksum_str) {
+             throw "Checksum record not found for file name '$basename' in file from url: '$url'"
+         }
+
+        $web_checksum_str_splitted = $web_checksum_str[0].ToString().split(" ", 2)
+        $hash_from_file = $web_checksum_str_splitted[0].Trim()
+        # Remove any non-alphanumeric characters
+        $hash_from_file = $hash_from_file -replace '\W+', ''
     } Catch [System.Net.WebException] {
         $module.Result.status_code = [int] $_.Exception.Response.StatusCode
         $module.FailJson("Error requesting '$url'. $($_.Exception.Message)", $_)
     } Catch {
-        $module.FailJson("Error when requesting 'Last-Modified' date from '$url'. $($_.Exception.Message)", $_)
+        $module.FailJson("Error get HASH data file from '$url'. $($_.Exception.Message)", $_)
+    }
+    Finally {
+        if($webResponse) {
+           $webResponse.Close()
+        }
     }
-    $module.Result.status_code = [int] $webResponse.StatusCode
-    $module.Result.msg = [string] $webResponse.StatusDescription
-    $webResponse.Close()
+    if ( [bool]([System.Uri]$url).isFile ) {
+        $module.Result.status_code = 200
+        $module.Result.msg = 'OK'
+    } else {
+        $module.Result.status_code = [int] $webResponse.StatusCode
+        $module.Result.msg = [string] $webResponse.StatusDescription
+    }
+    return $hash_from_file
+}
+
+Function Compare-ModifiedFile {
+    param($module, $url, $dest, $headers, $credentials, $timeout, $use_proxy, $proxy, $check_length)
+    if ($checksum) {
+        Try {
+            $is_modified_checksum = Compare-ModifiedChecksumFile -dest $dest -checksum $checksum
```

Comment:
I don't see the need to have this in a separate cmdlet. Just compare them here as `Compare-ModifiedChecksumFile` is not being used anywhere else

Reply:
Fixed